### DISCUSSION TITLE
Things I want to try in Identity Box

### DISCUSSION THREAD

**Discussion by marcinczenko** [identity-box/identity-box#74] [Ideas]
Here are the ideas I want to research and work on in the frame of the Identity Box project. The list will grow. I just flushed my mind:

> If you happen to be an investor believing in that idea - we need funding ;):

- extend hush-hush so that you can easily retrieve messages sent to you, grouped by your identities
- add ability to associate with more than one Identity Box
- finish adding a database as standard Identity Box service
- ESM only where possible
- improve linting (switch from standard to eslint)
- add basic verifiable credentials support (authentication SDK with SSI)
- update socket.io in rendezvous service to the new version
- update navigation in react native to the new version
- add notifications to the idapp so that the user is notified when an action is required
- further extend cli tools (next one will be automatically updating identity box services): https://github.com/identity-box/cli
- a way to automatically setup and backup the identity box
- automatic/convenient OS updates
- automatic/convenient identity box backup (now it is an easy, yet manual, process that requires screen attached)
- use Identity Box on some commercial websites to facilitate a <<send me a message>> functionality (like a communicator, but based on SSI and e2e encryption
- scripting for personal hosting
- github with frontend that will allow pull requests and CI pipelines but without the burden of GitHub
- a site to generate a passphrase (we already have a separate package with CLI that works nicely with npx: https://github.com/identity-box/mnemonic-passphrase)
- version of identity box without IPFS - how would that feel - what would work better, or worse, would that let us get faster to make something that can be reliable enough to use in production?
- blogging platform for identity box - lots of work is already there in Confluenza and Web Community blog
- see what kind of things can be ported to Remix - how would that feel over Next.js?
- check if pubsub and IPNS in IPFS works more reliably. Last time we checked that was on v0.4.0, now we are on version v0.11.0. If it works we can get rid of the hack where we keep the swarm of identity boxes active by running a timer service
- e-commerce platform for Identity box - with modern banking solutions, shopify is a thing of the past
- check if contentful is something for us
- see how easily would it be to build Identity App in a native way using official crypto from apple - how much effort it is? Maybe it is a way to get something published in the App Store. We are now using React Native (which works very good btw), we are using uniform crypto across all the platforms (tweetnacl), but this may generate some hurdless when publishing the app to the App Store
- build a version of the Identity App as a browser app. How can we make association with the box and communication with the box and the browser working without an ios app? This is challenging the current realization of rendezvous service
- password manager for identity box - before we will have SSI everywhere...
- authentication for identity box - again as a transition phase to SSI...
- yes, Diary App/community tool - we have designs and new insights - lets build it!
- parallel Internet - offline-first and radio communication instead of the Internet
- suggested by @mmzsource: investigate using [Shamir's Secret Sharing](https://en.wikipedia.org/wiki/Shamir%27s_Secret_Sharing) to eliminate the necessity to use passphrase for the backup. Example [Go implementation](https://github.com/jesseduffield/horcrux), [Javascript](https://www.npmjs.com/package/shamirs-secret-sharing).

**Comment by JeroenKnoops** [comment]
I would really like to see if we can recreate some of the functionality of [Krypton](https://krypt.co/).

We're using it to add 2FA on SSH-keys (and sites) and to sign Git Commits with PGP. Krypt.co has been bought by Akamai and the 2FA on sites are moved to a new application.
The 2FA on SSH-keys and signing of Git Commits with PGP are not. There is no real alternative right now and I think it will fit idBox perfectly. @brend-smits

**Reply by marcinczenko** [reply]
I remember you talking about it earlier, but I don't remember everything. After they were acquired, did they change something or get rid of some functionality?

**Reply by Brend-Smits** [reply]
This would be awesome to have. 
The functionality of signing Git Commits with PGP and having SSH Keys is still in Krypt.co but the software is no longer being updated and is essentially deprecated. They did not port this functionality to the Akami application. 
As Jeroen said, there are no easy-to-use alternatives. If this would be in ID-box, I would use it, as long as it's easy enough to set up.

**Comment by JeroenKnoops** [comment]
I think a lot of their stuff is open sourced: https://github.com/kryptco
